Lib/test/test_capi/test_long.py

```diff
@@ -34,35 +53,378 @@ def test_compact_known(self):
         self.assertEqual(_testcapi.call_long_compact_api(sys.maxsize),
                          (False, -1))
 
+    def test_long_check(self):
+        """Test PyLong_Check()"""
+        check = _testcapi.pylong_check
+        self.assertTrue(check(1))
+        self.assertTrue(check(123456789012345678901234567890))
+        self.assertTrue(check(-1))
+        self.assertTrue(check(True))
+        self.assertTrue(check(IntSubclass(1)))
+        self.assertFalse(check(1.0))
+        self.assertFalse(check(object()))
+        # CRASHES check(NULL)
+
+    def test_long_checkexact(self):
+        """Test PyLong_CheckExact()"""
+        check = _testcapi.pylong_checkexact
+        self.assertTrue(check(1))
+        self.assertTrue(check(123456789012345678901234567890))
+        self.assertTrue(check(-1))
+        self.assertFalse(check(True))
+        self.assertFalse(check(IntSubclass(1)))
+        self.assertFalse(check(1.0))
+        self.assertFalse(check(object()))
+        # CRASHES check(NULL)
+
+    def test_long_fromdouble(self):
+        """Test PyLong_FromDouble()"""
+        fromdouble = _testcapi.pylong_fromdouble
+        self.assertEqual(fromdouble(5.0), 5)
+        self.assertEqual(fromdouble(5.1), 5)
+        self.assertEqual(fromdouble(5.9), 5)
+        self.assertEqual(fromdouble(-5.1), -5)
+        self.assertEqual(fromdouble(-5.9), -5)
+        self.assertEqual(fromdouble(sys.float_info.max), int(sys.float_info.max))
+        self.assertEqual(fromdouble(-sys.float_info.max), -int(sys.float_info.max))
+        self.assertRaises(OverflowError, fromdouble, float('inf'))
+        self.assertRaises(OverflowError, fromdouble, float('-inf'))
+        self.assertRaises(ValueError, fromdouble, float('nan'))
+
+    def test_long_fromvoidptr(self):
+        """Test PyLong_FromVoidPtr()"""
+        fromvoidptr = _testcapi.pylong_fromvoidptr
+        obj = object()
+        x = fromvoidptr(obj)
+        y = fromvoidptr(NULL)
+        self.assertIsInstance(x, int)
+        self.assertGreaterEqual(x, 0)
+        self.assertIsInstance(y, int)
+        self.assertGreaterEqual(y, 0)
```

Comment:
NULL can be greater than 0?

```suggestion
        self.assertEqual(y, 0)
```

Reply:
Hmm. One part of me remember, that bitpattern of NULL is not always zeroes. But on other hand, a lot of code in CPython depends on this (starting from initializing pointers with `memset()`).